index.js

```diff
@@ -17,12 +17,25 @@ function shouldComponentUpdate(nextProps, nextState) {
   return !shallowEqual(this.props, nextProps) || !shallowEqual(this.state, nextState);
 }
 
+function getFunctionName(fun) {
+  var ret = fun.toString();
+  ret = ret.substr('function '.length);
+  ret = ret.substr(0, ret.indexOf('('));
+  return ret;
+}
+
 /**
  * Makes the given component "pure".
  *
  * @param object component Component.
  */
 function pureRenderDecorator(component) {
+  if (component.prototype.shouldComponentUpdate !== undefined) {
+    var componentName = component.prototype.name || component.name || getFunctionName(component);
```

Comment:
Yep, that's a good idea! I'll see if I can steal the getName function from React.